FPIS/Services/Product.Service.cs

```diff
@@ -0,0 +1,80 @@
+﻿using FPIS.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+namespace FPIS.Services
+{
+    /// <summary>
+    /// This class represents a service that works the `Products` models to perform
+    /// database transactions.
+    /// </summary>
+    public class ProductService
+    {
+        private readonly AppDbContext _dbContext;
+
+
+        /// <summary>
+        /// Constructor initialize a new `ProductService` with a DbContext
+        /// </summary>
+        /// <param name="dbContext"></param>
+        public ProductService(AppDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        /// <summary>
+        /// Checks if a product name exists already.
+        /// </summary>
+        /// <param name="productName"></param>
+        /// <returns></returns>
+        public bool DoesProductExists(string productName)
+        {
+            var product = _dbContext.Products.FirstOrDefault(p => p.ProductName.ToLower() == productName.ToLower());
+
+            if (product != null)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+
+        /// <summary>
+        /// Creates a new `Product` with a product name
+        /// </summary>
+        /// <param name="productName"></param>
+        public Product CreateProduct(string productName)
+        {
+            var analysisItem = _dbContext.AnalysisItems.Add(
+                    new()
+                    {
+                        ItemType = "Product"
+                    }
+                ).Entity;

```

Comment:
This feature is only meant to create a product :). I guess this entire block would be used elsewhere

Reply:
Looks fine now